client/app/components/app-header/AppHeader.jsx

```diff
@@ -0,0 +1,266 @@
+/* eslint-disable no-template-curly-in-string */
+
+import React, { useRef } from 'react';
+import { react2angular } from 'react2angular';
+
+import Dropdown from 'antd/lib/dropdown';
+import Button from 'antd/lib/button';
+import Icon from 'antd/lib/icon';
+import Menu from 'antd/lib/menu';
+import Input from 'antd/lib/input';
+import Tooltip from 'antd/lib/tooltip';
+
+import FavoritesDropdown from './components/FavoritesDropdown';
+import { HelpTrigger } from '@/components/HelpTrigger';
+import CreateDashboardDialog from '@/components/dashboards/CreateDashboardDialog';
+
+import { currentUser, Auth, clientConfig } from '@/services/auth';
+import { $location, $route } from '@/services/ng';
+import navigateTo from '@/services/navigateTo';
+import { Dashboard } from '@/services/dashboard';
+import { Query } from '@/services/query';
+import frontendVersion from '@/version.json';
+import logoUrl from '@/assets/images/redash_icon_small.png';
+
+import './AppHeader.less';
+
+function onSearch(q) {
+  $location.path('/queries').search({ q });
+  $route.reload();
+}
+
+function Desktop() {
```

Comment:
Although implicit (and it's not a blocker for this) I would prefer naming those `DesktopHeader` and `MobileHeader`.

Reply:
+1 for this :slightly_smiling_face: